st2reactor/st2reactor/container/process_container.py

```diff
@@ -280,7 +343,49 @@ def _stop_sensor_process(self, sensor_id, exit_timeout=5):
             # Process hasn't exited yet, forcefully kill it
             process.kill()
 
-        self._delete_sensors(sensor_id)
+        self._delete_sensor(sensor_id)
+
+    def _respawn_sensor(self, sensor_id, sensor, exit_code):
```

Comment:
This method is called inside a for loop and there is an eventlet.sleep(). Maybe the respawn should be called in a separate eventlet to prevent one sensor restart from blocking others.